### DISCUSSION TITLE
[QST]what is the meaning of RMM in configs.md

### DISCUSSION THREAD

**Discussion by YeahNew** [NVIDIA/spark-rapids#5376] [General] [answered]
**What is your question?**
I'm sorry I don't understand the meaning of RMM in configs.md, it seems to be a resource? At the same time, I found that Rmm.class in cudf-0.14-cuda10-2.jar is the class used to initialize GPU memory. Is there any connection between them?
When I set spark.rapids.memory.gpu.pool.enabled=true, the query#5 was executed for 1.2mins, when it was false, query#5 was executed for 1.6mins. This setting approximately accelerated the query by 33.3%. I want to know the internal reasons.
I also don't quite understand the explanation of this parameter in configs.md.
Can you help me?

**Comment by jlowe** [comment, marked as answer]
>  I don't understand the meaning of RMM 

RMM stands for RAPIDS Memory Manager.  It's designed to accelerate CUDA memory allocations.  Normally CUDA memory operations like `cudaMalloc` and `cudaFree` are very expensive because they need to update page tables on the device.  RMM allocates memory in large chunks and retains those allocations, turning them into a pool of memory that it then sub-allocates to satisfy memory requests.  This is much, much faster in practice than calling `cudaMalloc` and `cudaFree` directly.

> I also don't quite understand the explanation of this parameter in configs.md.

Note that this is covered in our [tuning guide](https://github.com/NVIDIA/spark-rapids/blob/branch-0.3/docs/tuning-guide.md#pooled-memory) along with other important settings for performance such as [using pinned host memory](https://github.com/NVIDIA/spark-rapids/blob/branch-0.3/docs/tuning-guide.md#pinned-memory).  If the explanation is still not clear after reading the tuning guide, please let us know and we'll work to improve the documentation.

**Comment by YeahNew** [comment]
Thank you, I have got it
